src/bio/bio_monitor.c

```diff
@@ -33,6 +33,62 @@
 /* Used to preallocate buffer to query error log pages from SPDK health info */
 #define DAOS_MAX_ERROR_LOG_PAGES 256
 
+/*
+ * Used for getting bio device state, which requires exclusive access from
+ * the device owner xstream.
+ */
+struct dev_state_msg_arg {
+	struct bio_xs_context	*xs;
+	struct bio_dev_state	*devstate;
+	bool			 send_msg_complete;
+	ABT_mutex		 mutex;
+	ABT_cond		 cond;
+};
+
+/* Copy out the bio_dev_state in the device owner xstream context */
+static void
+bio_get_dev_state_internal(void *msg_arg)
+{
+	struct dev_state_msg_arg	*dsm = msg_arg;
+
+	D_ASSERT(dsm != NULL);
+
+	ABT_mutex_lock(dsm->mutex);
+	dsm->devstate = &dsm->xs->bxc_blobstore->bb_dev_health.bdh_health_state;
```

Comment:
If we return a pointer instead of copy data back, other xstreams could access the bdh_health_state through the returned pointer while owner xstream updating it, so the reader may get inconsistent data. (reader read the data in bdh_health_state isn't atomic)